mlflow/tracking/client.py

```diff
@@ -1418,8 +1419,200 @@ def _is_plotly_figure(fig):
+        """
+        Logs an image in MLflow, supporting two use cases:
+
+        1. Time-Stepped Image Logging: Ideal for tracking changes or progressions through iterative
+            processes (e.g., during model training phases).
+            - Usage: `log_image(key, image, step, timestamp)`
+        2. Legacy Artifact File Image Logging: Best suited for static image logging where the image
+            is saved directly as a file artifact.
+            - Usage: `log_image(image, artifact_file)`
+
+        The following image formats are supported:
+        - `numpy.ndarray`_
+        - `PIL.Image.Image`_
+
+        .. _numpy.ndarray:
+            https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html
+
+        .. _PIL.Image.Image:
+            https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image
+
+        Numpy array support
+            - data type (( ) represents a valid value range):
+
+                - bool
+                - integer (0 ~ 255)
+                - unsigned integer (0 ~ 255)
+                - float (0.0 ~ 1.0)
+
+                .. warning::
+
+                    - Out-of-range integer values will be **clipped** to [0, 255].
+                    - Out-of-range float values will be **clipped** to [0, 1].
+
+            - shape (H: height, W: width):
+
+                - H x W (Grayscale)
+                - H x W x 1 (Grayscale)
+                - H x W x 3 (an RGB channel order is assumed)
+                - H x W x 4 (an RGBA channel order is assumed)
+
+        Args:
+            run_id: String ID of run.
+            key: Image name. This string may only contain alphanumerics, underscores
+                (_), dashes (-), periods (.), spaces ( ), and slashes (/).
+                All backend stores will support keys up to length 250, but some may
+                support larger keys.
+            image: The image object to be logged.
+            step: Integer training step (iteration) at which the image was saved.
+                Defaults to 0.
+            timestamp: Time when this image was saved. Defaults to the current system time.
+            artifact_file: Legacy. Specifies the path, in POSIX format, where the image
+                will be stored as an artifact relative to the run's root directory (for
+                example, "dir/image.png"). This parameter is kept for backward compatibility
+                and should not be used together with `key`, `step`, or `timestamp`.
+            *args: Positional arguments placeholder. Please refer to the documentation above.
+            **kwargs: Keyword arguments placeholder. Please refer to the documentation above.
+
+        .. code-block:: python
+            :caption: Time-stepped image logging numpy example
+
+            import mlflow
+            import numpy as np
+
+            image = np.random.randint(0, 256, size=(100, 100, 3), dtype=np.uint8)
+
+            run = client.create_run(experiment_id="0")
```

Comment:
This code will break if we test the docstring testing, since we have not created the `client` instance. shall we do the following instead? 

```
with mlflow.start_run() as run:
    client = mlflow.MlflowClient()
    client.log_image(...)
```